src/kili/adapters/kili_api_gateway/helpers/queries.py

```diff
@@ -57,6 +60,11 @@ def execute_query_from_paginated_call(
         )
         disable_tqdm = nb_elements_to_query is None or options.disable_tqdm
 
+        # If we need an unicity check, it means that new values can be inserted during the
+        # pagination process, so that the initial number of element to query cannot be trusted.
+        use_nb_elements_to_query = unicity_field is None or options.first is not None
+        unicity_values = []
```

Comment:
as unicity_values is just here to check the existence or not of the unicity value, it should be a dict to check it in less complexity